deployment/shell.nix

```diff
@@ -65,6 +66,39 @@ let
         ${terraform}/bin/terraform destroy ./terraform
       '';
 
+      # wait-github-status: wait until the current commit has been processed by hydra
+      # - checks the github status in a loop with 30s breaks until it is is "success"
+      waitGitHubStatus = writeShellScriptBin "wait-github-status" ''
+        set -eou pipefail
+
+        echo "[wait-github-status]: waiting for commit to get processed by hydra"
+
+        SLEEP_SECS=60
+        GH_STATUS=$(${hub}/bin/hub ci-status)
+
+        while [ $GH_STATUS != "success" ]; do
+         case "$GH_STATUS" in
+          failure|error|action_required|cancelled|timed_out)
+            echo "[wait-github-status]: $GH_STATUS"
+            exit 1
+            ;;
+          pending)
+            pritnf "."
+            sleep $SLEEP_SECS
+            GH_STATUS=$(${hub}/bin/hub ci-status)
```

Comment:
Are we going to get rate-limited doing this?

Reply:
I tried several invocations of `hub ci-status` in combination with querying the rate limits as specified in the GitHub API documentation:

```
curl \
  -H "Accept: application/vnd.github.v3+json" \
  https://api.github.com/rate_limit

{
  "resources": {
    "core": {
      "limit": 60,
      "remaining": 60,
      "reset": 1619170722,
      "used": 0
    },
    "graphql": {
      "limit": 0,
      "remaining": 0,
      "reset": 1619170722,
      "used": 0
    },
    "integration_manifest": {
      "limit": 5000,
      "remaining": 5000,
      "reset": 1619170722,
      "used": 0
    },
    "search": {
      "limit": 10,
      "remaining": 10,
      "reset": 1619167182,
      "used": 0
    }
  },
  "rate": {
    "limit": 60,
    "remaining": 60,
    "reset": 1619170722,
    "used": 0
  }
}
```

There were no changes. So I assume these calls don't get limited.